README.md

```diff
@@ -235,6 +236,120 @@ If you need the latest code from upstream git of OpenSCAP and/or
+| /tasks/<int:taskId>result/<int:resultId>/  | GET    | oscapd-cli result <taskId> <resultId> report | Gets <resultId> report for <taskId>. Note: HTML Output    |
+| /tasks/<int:taskId>/result/                | DELETE | oscapd-cli result <taskId> remove            | Removes all results for <taskId>                          |
+| /tasks/<int:taskId>/result/<int:resultId>/ | DELETE | oscapd-cli result <taskId> <resultId> remove | Removes <resultId> in <taskId>                            |
+| /tasks/<int:taskId>/run/                   | GET    | oscapd-cli task <taskId> run                 | Launch task with id <taskId>                              |
+| /tasks/<int:taskId>/<string:schedule>/     | PUT    | oscapd-cli task <taskId> enable/disable      | Enables/Disables <taskId>                                 |
+| /ssgs/                                     | GET    |                                              | Returns all SSGs installed on the system and its profiles |
+| /ssgs/                                     | POST   |                                              | Shows SSG's profiles within a json request                |
+
+### REST API Examples
+
+Get all tasks: 
+```
+curl -i http://127.0.0.1:5000/tasks/ -X GET
+```
+
+Create a new task:
+```
+newtask.json 
+{
+    "taskTitle":"New Task test",
+    "taskTarget":"localhost",
+    "taskSSG":"/usr/share/xml/scap/ssg/content/ssg-jre-ds.xml",
+    "taskTailoring":"",
+    "taskProfileId":"xccdf_org.ssgproject.content_profile_stig-java-upstream",
+    "taskOnlineRemediation":"1",
+    "taskScheduleNotBefore":"",
+    "taskScheduleRepeatAfter":""
+}
+  
+curl -i http://127.0.0.1:5000/tasks/ -X POST -H "Content-Type: application/json" -d '@newtask.json'
+```
+
+Update an existing task:   
+```
+updatetask.json
+{
+    "taskTitle":"New Task test modified",
+    "taskTarget":"localhost",
+    "taskSSG":"",
+    "taskTailoring":"",
+    "taskProfileId":"",
+    "taskOnlineRemediation":"yes",
+    "taskScheduleNotBefore":"",
+    "taskScheduleRepeatAfter":""
+}
+    
+curl -i http://127.0.0.1:5000/tasks/1/ -X PUT -H "Content-Type: application/json" -d '@updatetask.json'
+```
+
+Get info from existing task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/ -X GET
+``` 
+
+Get guide info from existing task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/guide/ -X GET
+```
+
+Get result from existing task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/result/1/ -X GET
+```
+
+Delete all results from existing task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/result/ -X DELETE
+```
+
+Delete result from existing task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/result/1/ -X DELETE
+```
+
+Launch existing task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/run/ -X GET
+```
+
+Enable/Disable existig task:
+```
+curl -i http://127.0.0.1:5000/tasks/1/enable/ -X PUT
+curl -i http://127.0.0.1:5000/tasks/1/disable/ -X PUT
+```
+
+Get all SSGs installed on the system:
+```
+curl -i http://127.0.0.1:5000/ssgs/ -X GET
```

Comment:
This  request gives me:
```
[vagrant@localhost vagrant]$ curl -i http://127.0.0.1:5000/ssgs/ -X GET
HTTP/1.0 404 NOT FOUND
Content-Type: text/html
Content-Length: 233
Server: Werkzeug/0.14.1 Python/3.7.0
Date: Mon, 05 Nov 2018 10:45:47 GMT

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server.  If you entered the URL manually please check your spelling and try again.</p>                                                                                  
[vagrant@localhost vagrant]$ 
```

I thought that because of missing slash in `rest.api.py` line 83.

But if I run `curl -i http://127.0.0.1:5000/ssgs -X GET`, I get  this:
```
[vagrant@localhost vagrant]$ curl -i http://127.0.0.1:5000/ssgs -X GET
HTTP/1.0 500 INTERNAL SERVER ERROR
Content-Type: text/html
Content-Length: 291
Server: Werkzeug/0.14.1 Python/3.7.0
Date: Mon, 05 Nov 2018 10:47:58 GMT

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request.  Either the server is overloaded or there is an error in the application.</p>
[vagrant@localhost vagrant]$ 
```

@mvazquezc Could you take a look at it, please?

Reply:
Hi @jan-cerny,

Thanks for the heads up. Two things happened:

1. The endpoint was not properly defined as you pointed out.
2. There was a race-condition causing tailoring-file to be set as a folder when no tailoring file was defined. 

I've pushed some changes that should take care of that. Thanks for the review!

Reply:
@mvazquezc great